core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java

```diff
@@ -430,56 +430,65 @@ void createMeasurementEvent(
     ) {
         writerWorkerId = LineTcpMeasurementEventType.ALL_WRITERS_INCOMPLETE_EVENT;
         final TableUpdateDetails.ThreadLocalDetails localDetails = tableUpdateDetails.getThreadLocalDetails(workerId);
-        final BoolList processedCols = localDetails.getProcessedCols();
-        final LowerCaseCharSequenceHashSet addedCols = localDetails.getAddedCols();
-        processedCols.setAll(localDetails.getColumnCount(), false);
-        addedCols.clear();
+        // tracking of processed columns by their index, duplicates will be ignored
+        final BoolList processedCols = localDetails.resetProcessedCols();
+        // tracking of processed columns by their name, duplicates will be ignored
+        // columns end up in this set only if their index cannot be resolved, i.e. new columns
+        final LowerCaseCharSequenceHashSet addedColsUtf16 = localDetails.resetAddedCols();
         this.tableUpdateDetails = tableUpdateDetails;
         long timestamp = parser.getTimestamp();
         if (timestamp != LineTcpParser.NULL_TIMESTAMP) {
             timestamp = timestampAdapter.getMicros(timestamp);
-            processedCols.setQuick(tableUpdateDetails.getTimestampIndex(), true);
         }
         long bufPos = bufLo;
         long bufMax = bufLo + bufSize;
         long timestampBufPos = bufPos;
-        //timestamp and entitiesWritten are saved to timestampBufPos after saving all fields
+        // timestamp and entitiesWritten are saved to timestampBufPos after saving all fields
+        // because their values are worked out while the columns are processed
         bufPos += Long.BYTES;
         int entitiesWritten = 0;
         bufPos += Integer.BYTES;
         for (int nEntity = 0, n = parser.getEntityCount(); nEntity < n; nEntity++) {
             if (bufPos + Long.BYTES < bufMax) {
                 LineTcpParser.ProtoEntity entity = parser.getEntity(nEntity);
-                int colIndex = localDetails.getColumnIndex(entity.getName());
+                int colIndex = localDetails.getColumnIndex(entity.getName(), parser.hasNonAsciiChars());
                 if (colIndex < 0) {
-                    final DirectByteCharSequence colName = entity.getName();
-                    if (!addedCols.add(colName)) {
+                    // column index not found, processing column by name
+                    CharSequence colNameUtf16 = utf8ToUtf16(entity.getName(), parser.hasNonAsciiChars());
```

Comment:
virtually every time we do not find index of column in UTF8 map, we are encoding UTF8-UTF16 twice now for the same string. May be there is a way to salvage the first attempt to encode and reuse ?

Reply:
moved duplicate column check inside index lookup, was planning to do it as the next step anyway.
this way we can avoid duplicate efforts.